### DISCUSSION TITLE
Supabase real time is failing because of ipv6 is disabled in my environment.

### DISCUSSION THREAD

**Discussion by arunbe1** [supabase/supabase#16416] [Questions]
I am trying to deploy supabase in our kube environment. Supabase real time is failing because of ipv6. Error message is below
16:19:49.035 [warning] [libcluster:fly6pn] dns polling strategy is selected, but query or basename param is invalid: %{node_basename: "realtime", query: nil}

16:19:49.043 [error] Failed to start Ranch listener RealtimeWeb.Endpoint.HTTP in :ranch_tcp:listen([{:cacerts, :...}, {:key, :...}, {:cert, :...}, :inet6, {:port, 9999}]) for reason :eafnosupport (address family not supported by protocol family)

 

16:19:49.046 [notice] Application realtime exited: Realtime.Application.start(:normal, []) returned an error: shutdown: failed to start child: RealtimeWeb.Endpoint

    ** (EXIT) shutdown: failed to start child: {:ranch_listener_sup, RealtimeWeb.Endpoint.HTTP}

        ** (EXIT) shutdown: failed to start child: :ranch_acceptors_sup

            ** (EXIT) {:listen_error, RealtimeWeb.Endpoint.HTTP, :eafnosupport}
Need solution to disable ipv6 in supabase?

**Comment by GaryAustin1** [comment]
There seems to be a configuration option for self hosted...
https://supabase.com/docs/guides/self-hosting/realtime/config
![image](https://github.com/supabase/supabase/assets/54564956/c99019aa-59a3-4581-b892-217574da0fd9)

Sorry I can't  help though on how to set that option.

**Reply by arunbe1** [reply]
@GaryAustin1 - As you mentioned already looked into that but not sure how to implement that. Need help to implement that value for real time.

**Reply by GaryAustin1** [reply]
Maybe this is a better description of the current self hosted realtime, but I'm not sure as I don't use it.
See All Relevant Options here:  https://github.com/supabase/realtime
`ERL_AFLAGS                 # {string}      Set to either "-proto_dist inet_tcp" or "-proto_dist inet6_tcp" depending on whether or not your network uses IPv4 or IPv6, respectively.`
Otherwise you have to see if someone who has done this comes along.

**Reply by arunbe1** [reply]
@GaryAustin1 - Thanks for your response. I tried as you mentioned still having same problem.

**Reply by GeoApopei18** [reply]
Any updates or info here? I'm self hosting on an infrastructure that doesn't allow ipv6 and would be very much appreciate a resolution to this thread, if any. 
The realtime image version I'm using: supabase/realtime:v2.25.60

Thank you!

**Reply by mihail-ciontu** [reply]
i have the same problem as the author and @GeoApopei18, is there a way to use realtime on a ipv4?
